### PR TITLE
Dof mgr projection test

### PR DESCRIPTION
**Pull request by mperego**
Modified Cartesian Connectivity Manager to allow tet meshes (created by splitting each Hex into 6 Tets) and triangular meshes (created by splitting each quad into two triangles).

Added test for L2 projection to test Panzer DofMgr, testing L2 projection on (continuous) HGrad/HCurl/HDiv and HVol spaces and for Hex/Tet/Quad/Triangle shapes.

@trilinos/panzer 

## Testing

The new tet and triangle meshes are indirectly tested in the added projection test.

### CODE REVIEW

**Comment by mperego** [comment]
@rppawlo @egphill here you can find a projection test for HDIV on a tet mesh.

**Comment by mperego** [comment]
@eric-c-cyr Can you check the modifications I made to the Cartesian Conn Manager to support Tet and Triangle meshes? (first commit)

File: packages/panzer/dof-mgr/test/fe_assembly/test_fe_projection.hpp [hunk 1]
```diff
@@ -0,0 +1,798 @@
...
+    //compute global ids of element vertices
+    DynRankViewGId ConstructWithLabel(elemNodesGID, numOwnedElems, numNodesPerElem);
+    {
+      for(int i=0; i<numOwnedElems; ++i) {
+        const auto GIDs = connManager->getConnectivity(i);
+        for(int j=0; j<numNodesPerElem; ++j) {
+          elemNodesGID(i,j) = GIDs[j];
+        }
+      }
+    }
```
Comment (rppawlo) [inline packages/panzer/dof-mgr/test/fe_assembly/test_fe_projection.hpp:483]: It looks like this relies on UVM?

**Review by rppawlo — APPROVED** [review]
Looks good to me! Thanks @mperego !

**Review by rppawlo — APPROVED** [review]

**Pull request merged**